bundles/org.openhab.binding.guntamatic/src/main/java/org/openhab/binding/guntamatic/internal/GuntamaticHandler.java

```diff
@@ -0,0 +1,426 @@
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    case CHANNEL_CONTROLWWHEAT0:
+                    case CHANNEL_CONTROLWWHEAT1:
+                    case CHANNEL_CONTROLWWHEAT2:
+                        param = getThing().getProperties().get(PARAMETER_WWHEAT).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    case CHANNEL_CONTROLEXTRAWWHEAT0:
+                    case CHANNEL_CONTROLEXTRAWWHEAT1:
+                    case CHANNEL_CONTROLEXTRAWWHEAT2:
+                        param = getThing().getProperties().get(PARAMETER_EXTRAWWHEAT).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    default:
+                        return;
+                }
+                String response = sendGetRequest(PARSET_URL, "syn=" + param + "&value=" + command.toString());
+                // logger.warn("{} syn={}&value={}", PARSET_URL, param, command.toString());
+                State newState = new StringType(response);
+                updateState(channelID, newState);
+            } else {
+                logger.warn("A 'key' needs to be configured in order to control the Guntamatic Heating System");
+            }
+        }
+    }
+
+    private void parseAndUpdate(String html) {
+        String[] daqdata = html.split("\\n");
+
+        for (Integer i : channels.keySet()) {
+            String channel = channels.get(i);
+            String unit = units.get(i);
+            if ((channel != null) && (unit != null)) {
+                String value = daqdata[i].trim();
+                Channel chn = thing.getChannel(channel);
+                if ((chn != null) && ("Switch".equals(chn.getAcceptedItemType()))) {
+                    // Guntamatic uses German OnOff when configured to German and English OnOff for all other languages
+                    value = value.replace("AUS", "OFF").replace("EIN", "ON");
+                }
+                State newState = new StringType(value + unit);
+                updateState(channel, newState);
+            } else {
+                logger.warn("Data for not intialized ChannelId '{}' received", i);
+            }
+            // TypeParser.parseState(this.acceptedDataTypes, sensorValue);
+        }
+    }
+
+    private void parseAndJsonInit(String html) {
+        try {
+            // remove non JSON compliant, empty element ",,"
+            JsonArray json = JsonParser.parseString(html.replace(",,", ",")).getAsJsonArray();
+            for (int i = 1; i < json.size(); i++) {
+                JsonObject points = json.get(i).getAsJsonObject();
+                int id = points.get("id").getAsInt();
+                String type = points.get("type").getAsString();
```

Comment:
I assume that in theory points.get("XXX") could return null. So maybe add a null check.

Reply:
```
            for (int i = 1; i < json.size(); i++) {
                JsonObject points = json.get(i).getAsJsonObject();
                if ( points.has("id") ) {
                    int id = points.get("id").getAsInt();
                    if ( points.has("type") ) {
                        String type = points.get("type").getAsString();
                        types.put(id, type);
                    }
                }
            }
```